lib/resty/redis.lua

```diff
@@ -107,7 +65,7 @@ end
 
 
 function _M.set_timeout(self, timeout)
-    local sock = self.sock
+    local sock = rawget(self, "sock")
```

Comment:
Okay, never mind. It seems that we do need the `rawget` to avoid triggering our own `__index` metamethod. Alas. Maybe always ignore keys starting with an underscore in the `__index` metamethod and turns `sock` into `_sock`, for example?